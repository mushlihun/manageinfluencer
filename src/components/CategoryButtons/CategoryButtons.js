import React from 'react'
import styled from 'styled-components/macro'
import fork from '../../icons/fork.svg'
import dumbbell from '../../icons/gym.svg'
import lipsticks from '../../icons/beauty.svg'
import mannequin from '../../icons/mannequin.svg'
import vase from '../../icons/vase.svg'
import plane from '../../icons/plane.svg'
import { Switch, Route, Link } from 'react-router-dom'

export default function CategoryButtons() {
  return (
    <SectionStyled>
      <Switch>
        <ContainerStyled>
          <section>
            <Route>
              <Link to="/food">
                <ButtonStyled category="food">
                  <IconsStyled src={fork} alt="fork and spoon" />
                  <H2Styled>Food</H2Styled>
                </ButtonStyled>
              </Link>
            </Route>
            <Route>
              <Link to="/fitness">
                <ButtonStyled category="fitness">
                  <IconsStyled src={dumbbell} alt="dumbbell" />
                  <H2Styled>Fitness</H2Styled>
                </ButtonStyled>
              </Link>
            </Route>
          </section>
          <section>
            <Route>
              <Link to="/beauty">
                <ButtonStyled category="beauty">
                  <img src={lipsticks} alt="lipsticks" />
                  <H2Styled>Beauty</H2Styled>
                </ButtonStyled>
              </Link>
            </Route>
            <Route>
              <Link to="/fashion">
                <ButtonStyled category="fashion">
                  <img src={mannequin} alt="fashion mannequin" />
                  <H2Styled>Fashion</H2Styled>
                </ButtonStyled>
              </Link>
            </Route>
          </section>
          <section>
            <Route>
              <Link to="/interior">
                <ButtonStyled category="interior">
                  <img src={vase} alt="vase" />
                  <H2Styled>Interior</H2Styled>
                </ButtonStyled>
              </Link>
            </Route>
            <Route>
              <Link to="/travel">
                <ButtonStyled category="travel">
                  <img src={plane} alt="plane" />
                  <H2Styled>Travel</H2Styled>
                </ButtonStyled>
              </Link>
            </Route>
          </section>
        </ContainerStyled>
      </Switch>
    </SectionStyled>
  )
}

const ContainerStyled = styled.section`
  margin-top: 80px;
`

const SectionStyled = styled.section`
  text-align: center;
  margin: 20px;
`

const ButtonStyled = styled.button`
  width: 130px;
  height: 130px;
  background: transparent;
  margin: 10px;
  border-radius: 10%;
  cursor: pointer;
  border-color: ${({ category }) =>
    category === 'food'
      ? 'var(--darkyellow)'
      : category === 'fitness'
      ? 'var(--lightpink)'
      : category === 'beauty'
      ? 'var(--lightgreen)'
      : category === 'fashion'
      ? 'var(--lightblue)'
      : category === 'interior'
      ? 'var(--lightpurple)'
      : 'var(--darkyellow)'};
`
const IconsStyled = styled.img`
  margin: 10px 0;
  vertical-align: top;
`

const H2Styled = styled.h2`
  color: var(--main-font-color);
  font-size: 20px;
  font-weight: var(--font-weight-light);
  vertical-align: bottom;
  margin-bottom: 7px;
  margin-top: 3px;
`
