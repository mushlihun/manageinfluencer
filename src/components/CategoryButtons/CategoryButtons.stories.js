import React from 'react'
import CategoryButtons from './CategoryButtons'

export default {
  title: 'Components/Button',
  component: CategoryButtons,
}

export const categories = () => {
  return <CategoryButtons />
}
