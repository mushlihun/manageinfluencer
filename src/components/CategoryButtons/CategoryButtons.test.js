import React from 'react'
import renderer from 'react-test-renderer'
import CategoryButtons from './CategoryButtons'
import '@testing-library/jest-dom/extend-expect'
import { MemoryRouter } from 'react-router-dom'

describe('CategoryButtons', () => {
  it('renders CategoryButtons correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <CategoryButtons />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
