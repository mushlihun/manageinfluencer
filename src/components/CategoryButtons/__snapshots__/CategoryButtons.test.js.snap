// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryButtons renders CategoryButtons correctly 1`] = `
<section
  className="CategoryButtons__SectionStyled-fe33yf-1 cjRsGv"
>
  <section
    className="CategoryButtons__ContainerStyled-fe33yf-0 ggAkgj"
  >
    <section>
      <a
        href="/food"
        onClick={[Function]}
      >
        <button
          className="CategoryButtons__ButtonStyled-fe33yf-2 dcyjSd"
        >
          <img
            alt="fork and spoon"
            className="CategoryButtons__IconsStyled-fe33yf-3 ZxEs"
            src="fork.svg"
          />
          <h2
            className="CategoryButtons__H2Styled-fe33yf-4 iidzug"
          >
            Food
          </h2>
        </button>
      </a>
      <a
        href="/fitness"
        onClick={[Function]}
      >
        <button
          className="CategoryButtons__ButtonStyled-fe33yf-2 hodJBn"
        >
          <img
            alt="dumbbell"
            className="CategoryButtons__IconsStyled-fe33yf-3 ZxEs"
            src="gym.svg"
          />
          <h2
            className="CategoryButtons__H2Styled-fe33yf-4 iidzug"
          >
            Fitness
          </h2>
        </button>
      </a>
    </section>
    <section>
      <a
        href="/beauty"
        onClick={[Function]}
      >
        <button
          className="CategoryButtons__ButtonStyled-fe33yf-2 hXuZFc"
        >
          <img
            alt="lipsticks"
            src="beauty.svg"
          />
          <h2
            className="CategoryButtons__H2Styled-fe33yf-4 iidzug"
          >
            Beauty
          </h2>
        </button>
      </a>
      <a
        href="/fashion"
        onClick={[Function]}
      >
        <button
          className="CategoryButtons__ButtonStyled-fe33yf-2 gAUJfN"
        >
          <img
            alt="fashion mannequin"
            src="mannequin.svg"
          />
          <h2
            className="CategoryButtons__H2Styled-fe33yf-4 iidzug"
          >
            Fashion
          </h2>
        </button>
      </a>
    </section>
    <section>
      <a
        href="/interior"
        onClick={[Function]}
      >
        <button
          className="CategoryButtons__ButtonStyled-fe33yf-2 eJpUsJ"
        >
          <img
            alt="vase"
            src="vase.svg"
          />
          <h2
            className="CategoryButtons__H2Styled-fe33yf-4 iidzug"
          >
            Interior
          </h2>
        </button>
      </a>
      <a
        href="/travel"
        onClick={[Function]}
      >
        <button
          className="CategoryButtons__ButtonStyled-fe33yf-2 dcyjSd"
        >
          <img
            alt="plane"
            src="plane.svg"
          />
          <h2
            className="CategoryButtons__H2Styled-fe33yf-4 iidzug"
          >
            Travel
          </h2>
        </button>
      </a>
    </section>
  </section>
</section>
`;
